sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/channel/ManagedChannelFactory.java

```diff
@@ -70,13 +92,42 @@ public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
    */
   private static class Default extends ManagedChannelFactory {
     @Override
-    public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
+    public ManagedChannel forDescriptor(
+        ApiServiceDescriptor apiServiceDescriptor, List<ClientInterceptor> interceptors) {
       return ManagedChannelBuilder.forTarget(apiServiceDescriptor.getUrl())
           .usePlaintext(true)
           // Set the message size to max value here. The actual size is governed by the
           // buffer size in the layers above.
           .maxInboundMessageSize(Integer.MAX_VALUE)
+          .intercept(interceptors)
           .build();
     }
   }
+
+  private static class InterceptedManagedChannelFactory extends ManagedChannelFactory {
+    private final ManagedChannelFactory channelFactory;
+    private final List<ClientInterceptor> interceptors;
+
+    private InterceptedManagedChannelFactory(
+        ManagedChannelFactory managedChannelFactory, List<ClientInterceptor> interceptors) {
+      this.channelFactory = managedChannelFactory;
+      this.interceptors = interceptors;
+    }
+
+    @Override
+    protected ManagedChannel forDescriptorOnly(ApiServiceDescriptor apiServiceDescriptor) {
+      return forDescriptor(apiServiceDescriptor, interceptors);
+    }
+
+    @Override
+    protected ManagedChannel forDescriptor(
+        ApiServiceDescriptor apiServiceDescriptor, List<ClientInterceptor> interceptors) {
+      return channelFactory.forDescriptor(apiServiceDescriptor, interceptors);
```

Comment:
The behavior in not well defined.
Though this is an intercepted factory, the original interceptors (interceptors from constructor) are not applied.